controllers/clustergroupupgrade_controller.go

```diff
@@ -2087,6 +2090,21 @@ func (r *ClusterGroupUpgradeReconciler) handleCguFinalizer(
 	return utils.DontReconcile, nil
 }
 
+func (r *ClusterGroupUpgradeReconciler) getCGUControllerWorkerCount() (count int) {
+	maxConcurrency, isSet := os.LookupEnv(utils.CGUControllerWorkerCountEnv)
+	var err error
+	if isSet {
+		count, err = strconv.Atoi(maxConcurrency)
+		if err != nil || count < 1 {
```

Comment:
Might be worthwhile to also check the upper limit and impose some restrictions in order to avoid resource hogging...

Reply:
This is mainly for internal use, e.g. experimenting in scale lab.

Reply:
k8s controller code probably handles 0 and negative values already. We are really stretching here to do all these checks just for logging purposes.

Reply:
https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/controller/controller.go#L130
True, true, but I think it's still nice to have this value set to what we need instead of the defaults set my controller-runtime :thinking:

Reply:
Yes. Again, it comes down to who will be setting this and how often. I should have made it clear that it will be really rare.